jdi-light-bootstrap-tests/src/test/java/io/github/epam/bootstrap/tests/composite/section/card/CardNavigationTests.java

```diff
@@ -16,16 +26,122 @@
+    private final String subtitleText = "With supporting text below as a natural lead-in to additional content.";
+    private final String buttonText = "Click Me!";
+    private final String activeLinkText = "Active";
+    private final String jdiLinkText = "JDI";
+    private final String disabledLinkText = "Disabled";
+    private final String activeLinkRef = "javascript: void()";
+    private final String jdiLinkRef = "https://github.com/jdi-testing/jdi-light";
+    private final String disabledLinkRef = "javascript: void()";
+    private final String jdiLinkPath = "/jdi-testing/jdi-light";
+    private final String activeLinkAlert = "Active Tab Clicked!";
+
     @BeforeMethod
     public void before() {
         shouldBeLoggedIn();
         bsPage.shouldBeOpened();
     }
 
     @Test
-    public void navLinkClickTest() {
-        cardNavigation.menu.select("Active");
-        cardNavigation.menu.select("JDI");
+    public void getTitleTextTest() {
+        assertEquals(cardNavigation.title.getText(), titleText);
+    }
+
+    @Test
+    public void getDescriptionTextTest() {
+        assertEquals(cardNavigation.subtitle.getText(), subtitleText);
+    }
+
+    @Test
+    public void getButtonTextTest() {
+        assertEquals(cardNavigation.button.getText(), buttonText);
+    }
+
+    @Test
+    public void getLinkTextTest() {
+        cardNavigation.nav.highlight();
+        assertEquals(cardNavigation.activeLink.getText(), activeLinkText);
+        assertEquals(cardNavigation.jdiLink.getText(), jdiLinkText);
+        assertEquals(cardNavigation.disabledLink.getText(), disabledLinkText);
+    }
+
+    @Test
+    public void getLinkValueTest() {
+        cardNavigation.nav.highlight();
+        assertEquals(cardNavigation.activeLink.getValue(), activeLinkText);
+        assertEquals(cardNavigation.jdiLink.getValue(), jdiLinkText);
+        assertEquals(cardNavigation.disabledLink.getValue(), disabledLinkText);
+    }
+
+    @Test
+    public void getLinkRefTest() {
+        cardNavigation.nav.highlight();
+        assertEquals(cardNavigation.activeLink.ref(), activeLinkRef);
+        assertEquals(cardNavigation.jdiLink.ref(), jdiLinkRef);
+        assertEquals(cardNavigation.disabledLink.ref(), disabledLinkRef);
+    }
+
+    @Test
+    public void getLinkUrlTest() {
+        cardNavigation.nav.highlight();
+        assertEquals(cardNavigation.jdiLink.url().getPath(), jdiLinkPath);
+    }
+
+    @Test
+    public void linkClickTest() {
+        cardNavigation.nav.highlight();
+
+        cardNavigation.activeLink.click();
+        validateAlert(is(activeLinkAlert));
+
+        cardNavigation.jdiLink.click();
+        ArrayList<String> tabs = new ArrayList<>(WebDriverFactory.getDriver().getWindowHandles());
+        WebDriver driver = WebDriverFactory.getDriver();
+        driver.switchTo().window(tabs.get(1));
+        assertEquals(getUrl() , jdiLinkRef);
+        driver.close();
+        driver.switchTo().window(tabs.get(0));
```

Comment:
I found out that we already have WindowsManager that already use index 1 as first index instead of 0, so I think we should use that class instead of direct calls to driver

Reply:
Done